### ISSUE TITLE
Support ::StringArray as a return type

### ISSUE THREAD

**Issue by plicease** [plicease/FFI-Platypus-Type-StringArray#1]
variable length string arrays will need to be NULL terminated, but fixed length string arrays should not have that requirement.

**Comment by yanick** [comment]
Hi! Just wondering if there is hope to have arrays of strings as return values? I'm trying to convert `Term::Caca` to FFI, and *of course* the very first function I'm looking at has that kind of return value. :-)

**Comment by plicease** [comment]
@yanick I opened this hoping that someone might implement it for me :) or if I found time I might do it myself.  A little time has since passed.  I just had a crack at it and have it implemented for fixed length arrays.  `NULL` terminated arrays may be more difficult.  Out of curiosity, what is the signature of the function you are trying to write a binding for.

**Comment by plicease** [comment]
Okay, implemented for both fixed-length and null terminated now.  I just released as `0.02`.  Please let me know if that helps out out!

**Comment by yanick** [comment]
It works beautifully! \o/ Thanks!

And the signature of the function is

```
/** \brief Get available display drivers
 *
 *  Return a list of available display drivers. The list is a NULL-terminated
 *  array of strings, interleaving a string containing the internal value for
 *  the display driver, and a string containing the natural language
 *  description for that driver.
 *
 *  This function never fails.
 *
 *  \return An array of strings.
 */
char const * const * caca_get_display_driver_list(void)
```

from https://github.com/cacalabs/libcaca/blob/master/caca/caca.c#L144
